pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -228,13 +265,35 @@ func (t *TopologyCache) SetNodes(nodes []*v1.Node) {
 	}
 }
 
+// HasPopulatedHints checks whether there are populated hints for a given service in the cache.
+func (t *TopologyCache) HasPopulatedHints(serviceKey string) bool {
+	return t.hintsPopulatedByService.Has(serviceKey)
+}
+
 // getAllocations returns a set of minimum and maximum allocations per zone. If
 // it is not possible to provide allocations that are below the overload
 // threshold, a nil value will be returned.
-func (t *TopologyCache) getAllocations(numEndpoints int) map[string]Allocation {
-	if t.cpuRatiosByZone == nil || len(t.cpuRatiosByZone) < 2 || len(t.cpuRatiosByZone) > numEndpoints {
-		klog.V(2).Infof("Insufficient info to allocate endpoints (%d endpoints, %d zones)", numEndpoints, len(t.cpuRatiosByZone))
-		return nil
+func (t *TopologyCache) getAllocations(numEndpoints int) (map[string]Allocation, *EventBuilder) {
+	if !t.sufficientNodeInfo {
+		return nil, &EventBuilder{
+			EventType: corev1.EventTypeWarning,
+			Reason:    "TopologyAwareHintsDisabled",
+			Message:   "Insufficient Node information: allocatable CPU or zone not specified on one or more nodes",
+		}
+	}
+	if len(t.cpuByZone) < 2 {
+		return nil, &EventBuilder{
+			EventType: corev1.EventTypeWarning,
+			Reason:    "TopologyAwareHintsDisabled",
+			Message:   "Nodes only ready in one zone",
+		}
+	}
+	if len(t.cpuRatiosByZone) > numEndpoints {
+		return nil, &EventBuilder{
+			EventType: corev1.EventTypeWarning,
+			Reason:    "TopologyAwareHintsDisabled",
+			Message:   fmt.Sprintf("Insufficient info to allocate endpoints (%d endpoints, %d zones)", numEndpoints, len(t.cpuRatiosByZone)),
```

Comment:
@robscott This one is insufficient **number** of endpoints right? 
Ditto for line 319?

Reply:
Yes, but can be more specific - There are less endpoints than there are zones